sources/packages/backend/libs/utilities/src/logger/logger.service.ts

```diff
@@ -1,11 +1,49 @@
 import { Injectable, ConsoleLogger, Scope } from "@nestjs/common";
 import { LogLevels, ProcessSummary } from "./process-summary";
+import { StringBuilder } from "@sims/utilities/string-builder";
+import { parseJSONError } from "@sims/utilities/parse-json";
 
 /**
  * Common log across entire solution.
  */
 @Injectable({ scope: Scope.TRANSIENT })
 export class LoggerService extends ConsoleLogger {
+  /**
+   * Log a friendly error message.
+   * @param message friendly message.
+   */
+  error(message: string): void;
+  /**
+   * Converts the object into a string and log it.
+   * @param object object to be logged.
+   */
+  error(object: unknown): void;
+  /**
+   * Log the error message and the error details.
+   * @param message friendly message.
+   * @param error error captured.
+   * @param context optional log context.
+   */
+  error(message: string, error: unknown, context?: string): void;
+  /**
+   * Log the error message and the error details.
+   * @param message friendly message or object to be logged.
+   * @param error error captured.
+   * @param context optional log context.
+   */
+  error(message: unknown, error?: unknown, context?: string): void {
+    const errorBuilder = new StringBuilder();
+    if (typeof message === "string") {
+      errorBuilder.appendLine(message);
+    } else {
+      errorBuilder.appendLine(parseJSONError(message));
+    }
+    if (error) {
+      errorBuilder.appendLine(parseJSONError(error));
+    }
+    super.error(errorBuilder.toString(), context);
```

Comment:
Do we want to call `super` if the context is not present. Should we wrap it in the `if` block like we are doing for `error`?

Reply:
From what I understood, the super class method would not be called when there is an implementation in the derived class unless we explicitly call it with `super`.

Reply:
In this case, we are "overriding" the base class method. We need the super to call the base method.

Reply:
Do you mean that we want to call the superclass method in all scenarios irrespective of the `context` being present or not?

Reply:
Thank You for the explanation on this.